### PR TITLE
Use batch command internally

### PR DESCRIPTION
**Pull request by t-woerner**
**IPAAnsibleModule: Add support for batch command in execute_ipa_commands**

    The method execute_ipa_commands has been extended to handle multi
    commands with the batch command.
    
    New constants for execute_ipa_commands debugging:
    
        DEBUG_COMMAND_ALL = 0b1111
        DEBUG_COMMAND_LIST = 0b0001
            Print the while command list
        DEBUG_COMMAND_COUNT = 0b0010
            Print the command number
        DEBUG_COMMAND_BATCH = 0b0100
            Print information about the batch slice size and currently executed
            batch slice
    
    New parameters have been added to execute_ipa_commands:
    
        batch: bool
            Enable batch command use to speed up processing
        batch_slice_size: integer
            Maximum mumber of commands processed in a slice with the batch
            command
        keeponly: list of string
            The attributes to keep in the results returned.
            Default: None (Keep all)
        debug: integer
            Enable debug output for the exection using DEBUG_COMMAND_*
    
    Batch mode can be enabled within the module with setting batch to True
    for execute_ipa_commands.
    
    Fixes: #1128 (batch command support)

**ipauser: Enable batch command use with keeponly**

    The use of the batch command is enabled for execute_ipa_commands.
    
    Additionally keeponly is set to ["randompassword"] as this is the only
    parameter that is used from the data returned from the commands.

**ipaservice: Enable batch command use with keeponly**

    The use of the batch command is enabled for execute_ipa_commands.
    
    Additionally keeponly is set to [] as nothing is used from the data returned
    from the commands.

**ipagroup: Enable batch command use with keeponly**

    The use of the batch command is enabled for execute_ipa_commands.
    
    Additionally keeponly is set to [] as nothing is used from the data returned
    from the commands.

**ipahost: Enable batch command with keeponly**

    The use of the batch command is enabled for execute_ipa_commands.
    
    Additionally keeponly is set to ["randompassword"] as this is the only
    parameter that is used from the data returned from the commands.

### CODE REVIEW

File: plugins/module_utils/ansible_freeipa_module.py [hunk 1]
```diff
@@ -1399,34 +1410,94 @@ def exception_handler(module, ex, exit_args, one_name):
                 if "errors" in argspec.args:
                     handlers_user_args["errors"] = _errors
 
+        debug = 0x10
```
Comment (rjeffman) [inline plugins/module_utils/ansible_freeipa_module.py:1413]: This should be removed.

File: plugins/module_utils/ansible_freeipa_module.py [hunk 2]
```diff
@@ -1399,34 +1410,94 @@ def exception_handler(module, ex, exit_args, one_name):
...
+        if debug & 0x1:
```
Comment (rjeffman) [inline plugins/module_utils/ansible_freeipa_module.py:1414]: What about calling these flags `IPA_DBG_CMD_LIST` and `IPA_DBG_CMD_COUNT` and make them class members?

Comment (t-woerner) [reply plugins/module_utils/ansible_freeipa_module.py]: I added DEBUG_COMMAND_ constants for this.

**Comment by rjeffman** [comment]
Changes look good to me. The pylint failure is being addressed in another PR.

I'll merge this PR as soon as checks pass.

**Review by rjeffman — APPROVED** [review]
LGTM.

**Pull request merged**